modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc

```diff
@@ -0,0 +1,74 @@
+// Module included in the following assemblies:
+//
+// * monitoring/accessing-third-party-monitoring-apis.adoc
+
+:_content-type: PROCEDURE
+[id="monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_{context}"]
+= Querying metrics by using the federation endpoint for Prometheus
+
+You can use the federation endpoint to query a cluster's scraped metrics from a network location outside the cluster. 
+To do so, access the Prometheus `/federate` endpoint for the cluster via an {product-title} route. 
+
+You can access the `/federate` endpoints for default platform monitoring and for user-defined projects.
+
+[WARNING]
+====
+A delay in retrieving metrics data occurs when you use federation. 
+This delay can affect the accuracy and timeliness of the scraped metrics.
+
+Using the federation endpoint can also degrade the performance and scalability of your cluster, especially if you use the federation endpoint  to retrieve large amounts of metrics data.
+To avoid these issues, follow these recommendations:
+
+* Do not try to retrieve all metrics data via the federation endpoint.
+Query it only when you want to retrieve a limited, aggregated data set. 
+For example, retrieving fewer than 1,000 samples per request helps minimize the risk of performance degradation. 
+
+* Avoid querying the federation endpoint frequently. 
+Limit queries to a maximum of one every 30 seconds.
+
+If you need to forward large amounts of data outside the cluster, use remote write instead. See the _Configuring remote write storage_ section for more information.
+====
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+* You have obtained the URL for the federation endpoint.
+* You have access to the cluster as a user with the `cluster-monitoring-view` role.
+* You have obtained a bearer token with get permission on the `namespaces` resource.
++
+[NOTE]
+====
+You can only use bearer token authentication to access the federation endpoint.
+====
+
+.Procedure
+
+. Retrieve the bearer token:
++
+[source,terminal]
+----
+$ token=`oc whoami -t`
+----
++
+. Query metrics from the `/federate` endpoint. 
+The following example queries `up` metrics:
++
+[source,terminal]
+----
+$ curl -G -s -k -H "Authorization: Bearer $token" \
+    'https:/<federation_endpoint_URL>/federate' \ <1>
+    --data-urlencode 'match[]=up'
+----
++
+<1> For <federation_endpoint_URL>, substitute the URL for the federation endpoint.
```

Comment:
```suggestion
<1> For <federation_host>, substitute the federation route's host.
```

Reply:
Changed to `For <federation_host>, substitute the host URL for the federation route`